src/sentry/api/endpoints/event_attachment_details.py

```diff
@@ -57,22 +56,17 @@ class EventAttachmentDetailsEndpoint(ProjectEndpoint):
     permission_classes = (EventAttachmentDetailsPermission,)
 
     def download(self, attachment):
-        file = File.objects.get(id=attachment.file_id)
-
-        content_type = attachment.content_type or file.headers.get(
-            "content-type", "application/octet-stream"
-        )
-        size = attachment.size or file.size
+        fp = attachment.getfile()
+        name = posixpath.basename(" ".join(attachment.name.split()))
 
-        fp = file.getfile()
-        response = StreamingHttpResponse(
-            iter(lambda: fp.read(4096), b""),
-            content_type=content_type,
+        response = FileResponse(
+            fp,
+            as_attachment=True,
+            filename=name,
+            content_type=attachment.content_type,
         )
+        response["Content-Length"] = attachment.size
```

Comment:
Why is this not a `StreamingHttpResponse` any more? Couldn't the file still be larger than `4096`?

Reply:
According to the [docs](https://docs.djangoproject.com/en/5.0/ref/request-response/#fileresponse-objects), this is a subclass of `StreamingHttpResponse` and streams the file in small chunks.
The reason I switched over was that this seems to automatically `close` the file. I had problems in tests related to that, and I have no idea how to properly do that with `StreamingHttpResponse`.

Reply:
> this is a subclass of StreamingHttpResponse

Nice, I did not know that! With `StreamingHttpResponse` you could probably use `fp` as a context manager to auto close it.